lib/src/utils.dart

```diff
@@ -58,9 +67,13 @@ Patch patchDelete(FileSpan span, {int start = 0, int end}) {
       span.file.span(span.start.offset + start, span.start.offset + end), "");
 }
 
-/// Emits a warning with [message] and [context];
-void emitWarning(String message, FileSpan context) {
-  print(context.message("WARNING - $message"));
+/// Emits a warning with [message] and optionally [context];
+void emitWarning(String message, {FileSpan context}) {
```

Comment:
This is the sort of API where I'd lean towards an optional rather than a named parameter. You almost always want to pass a span, it's unlikely that you'll need other optional parameters later, and it's a private API so if you do you can change it to a named parameter then without breaking anything.